### PR TITLE
Admin event plus resources 

### PR DESCRIPTION
**Pull request by gloveboxes**
Resource selection is now just part of adding or editing an event rather than separate ux
Clean up of executing pg functions - reader was now being closed. Just cleaner implementation
Added field length info to event form
Added clearer helper text to event form

### CODE REVIEW

File: src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs [hunk 1]
```diff
@@ -20,8 +19,11 @@ public partial class EventEditor : ComponentBase
 
     private bool isSubmitting = false;
 
+    private bool modelsUpdating = false;
```
Comment (aaronpowell) [inline src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs:22]: What's this used for?

File: src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs [hunk 2]
```diff
@@ -20,8 +19,11 @@ public partial class EventEditor : ComponentBase
...
+
     private IEnumerable<TimeZoneInfo>? TimeZones { get; set; }
```
Comment (aaronpowell) [inline src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs:25]:
```suggestion
```

File: src/AzureOpenAIProxy.Management/Components/Pages/EventAdd.razor.cs [hunk 3]
```diff
@@ -14,12 +11,34 @@ public partial class EventAdd : ComponentBase
...
+    [Inject]
+    public IModelService ModelService { get; set; } = null!;
+
+    public EventEditorModel Model { get; set; } = new();
+
+    protected override async Task OnInitializedAsync()
+    {
+        Model.AvailableModels = await ModelService.GetOwnerCatalogsAsync();
+        (Model.OrganizerEmail, Model.OrganizerName) = await AuthService.GetCurrentUserEmailNameAsync();
+        Model.Start = DateTime.Today.AddHours(-12); // allow for tz relative to utc
```
Comment (aaronpowell) [inline src/AzureOpenAIProxy.Management/Components/Pages/EventAdd.razor.cs:29]: How do we know that 12 is always relative to UTC?

File: src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs (hunk 1 above)
Comment (gloveboxes) [reply src/AzureOpenAIProxy.Management/Components/EventManagement/EventEditor.razor.cs:22]: Well spotted, no longer required...

File: src/AzureOpenAIProxy.Management/Components/Pages/EventAdd.razor.cs (hunk 3 above)
Comment (gloveboxes) [reply src/AzureOpenAIProxy.Management/Components/Pages/EventAdd.razor.cs:29]: // The container time in UTC 0. 
        // Start date should be yesterday in all time zones, end date should be 7 days from now.
        Model.Start = DateTime.Today.AddDays(-1);

**Review by aaronpowell — APPROVED** [review]

**Pull request merged**